src/core/template-wrapper.ts

```diff
@@ -12,28 +12,38 @@ interface ITemplateWrapperProps {
     key: string;
 }
 
+interface ITemplateWrapperState {
+    removalListenerRequired: boolean;
+}
+
 type TemplateWrapperRenderer = () => TemplateWrapper;
 
-class TemplateWrapper extends React.PureComponent<ITemplateWrapperProps> {
+class TemplateWrapper extends React.PureComponent<ITemplateWrapperProps, ITemplateWrapperState> {
     private readonly _removalListenerRef = React.createRef<HTMLElement>();
 
+    private get _listenerElement(): HTMLElement {
+        return this._removalListenerRef.current as HTMLElement;
+    }
+
     constructor(props: ITemplateWrapperProps) {
         super(props);
+
+        this.state = { removalListenerRequired: false };
+
+        this._onDxRemove = this._onDxRemove.bind(this);
     }
 
     public render() {
+        const removalListener = this.state.removalListenerRequired
+            ? React.createElement("span", { style: { display: "none" }, ref: this._removalListenerRef })
```

Comment:
Better make it constant between `render` calls?
```suggestion
            ? React.createElement("span", { style: removalListenerStyle, ref: this._removalListenerRef })
```